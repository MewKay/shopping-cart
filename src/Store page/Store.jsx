import { Link } from "react-router-dom";
import styles from "./Store.module.css";

const Store = function createStoreComponent() {
  return (
    <main className={styles["container"]}>
      <div className={styles["links-container"]}>
        <h2 className={styles["heading"]}>View Our Products</h2>
        <div className={styles["category-container"]}>
          <Link to={"men's clothing"} className={styles["category-link"]}>
            Men&apos;s Clothing
          </Link>
          <Link to={"women's clothing"} className={styles["category-link"]}>
            Women&apos;s Clothing
          </Link>
          <Link to={"electronics"} className={styles["category-link"]}>
            Electronics
          </Link>
          <Link to={"jewelery"} className={styles["category-link"]}>
            Jewelery
          </Link>
        </div>
        <Link to={"all"} className={styles["all-link"]}>
          View All
        </Link>
      </div>
    </main>
  );
};

export default Store;
