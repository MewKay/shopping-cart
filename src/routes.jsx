import Home from "./Home";
import Layout from "./Layout";

const routes = [
  {
    path: "/",
    element: <Layout />,
    children: [{ index: true, element: <Home /> }],
  },
];

export default routes;
