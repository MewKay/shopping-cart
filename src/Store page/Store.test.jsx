import { render, screen } from "@testing-library/react";
import {
  BrowserRouter,
  createMemoryRouter,
  RouterProvider,
  useParams,
} from "react-router-dom";
import { describe, expect, it, vi } from "vitest";
import Store from "./Store";
import routes from "../routes";
import userEvent from "@testing-library/user-event";

vi.mock("./Products", () => ({
  default: function MockProducts() {
    const { category } = useParams();

    switch (category) {
      case "men's clothing":
        return <>This is men clothing</>;
      case "women's clothing":
        return <>This is women clothing</>;
      case "electronics":
        return <>This is electronics</>;
      case "jewelery":
        return <>This is jewelery</>;
      default:
        return <>This is all</>;
    }
  },
}));

describe("Store component", () => {
  it("renders correctly", () => {
    const { container } = render(<Store />, { wrapper: BrowserRouter });

    expect(container).toMatchSnapshot();
  });

  describe("routes to the right Products category when corresponding link is clicked", () => {
    it("routes to men's clothing", async () => {
      const router = createMemoryRouter(routes, { initialEntries: ["/store"] });
      render(<RouterProvider router={router} />);

      const menClothLink = screen.getByRole("link", { name: "Men's Clothing" });
      await userEvent.click(menClothLink);

      const menClothText = await screen.findByText("This is men clothing");

      expect(menClothText).toBeInTheDocument();
    });

    it("routes to women's clothing", async () => {
      const router = createMemoryRouter(routes, { initialEntries: ["/store"] });
      render(<RouterProvider router={router} />);

      const womenClothLink = screen.getByRole("link", {
        name: "Women's Clothing",
      });
      await userEvent.click(womenClothLink);

      const womenClothText = await screen.findByText("This is women clothing");

      expect(womenClothText).toBeInTheDocument();
    });

    it("routes to electronics", async () => {
      const router = createMemoryRouter(routes, { initialEntries: ["/store"] });
      render(<RouterProvider router={router} />);

      const electronicsLink = screen.getByRole("link", { name: "Electronics" });
      await userEvent.click(electronicsLink);

      const electronicsText = await screen.findByText("This is electronics");

      expect(electronicsText).toBeInTheDocument();
    });

    it("routes to jewelery", async () => {
      const router = createMemoryRouter(routes, { initialEntries: ["/store"] });
      render(<RouterProvider router={router} />);

      const jeweleryLink = screen.getByRole("link", { name: "Jewelery" });
      await userEvent.click(jeweleryLink);

      const jeweleryText = await screen.findByText("This is jewelery");

      expect(jeweleryText).toBeInTheDocument();
    });

    it("routes to show all products", async () => {
      const router = createMemoryRouter(routes, { initialEntries: ["/store"] });
      render(<RouterProvider router={router} />);

      const viewAllLink = screen.getByRole("link", { name: "View All" });
      await userEvent.click(viewAllLink);

      const allText = await screen.findByText("This is all");

      expect(allText).toBeInTheDocument();
    });
  });
});
