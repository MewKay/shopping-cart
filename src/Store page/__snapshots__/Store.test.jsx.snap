// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Store component > renders correctly 1`] = `
<div>
  <main
    class="_container_5d4f08"
  >
    <div
      class="_links-container_5d4f08"
    >
      <h2
        class="_heading_5d4f08"
      >
        View Our Products
      </h2>
      <div
        class="_category-container_5d4f08"
      >
        <a
          class="_category-link_5d4f08"
          href="/men's clothing"
        >
          Men's Clothing
        </a>
        <a
          class="_category-link_5d4f08"
          href="/women's clothing"
        >
          Women's Clothing
        </a>
        <a
          class="_category-link_5d4f08"
          href="/electronics"
        >
          Electronics
        </a>
        <a
          class="_category-link_5d4f08"
          href="/jewelery"
        >
          Jewelery
        </a>
      </div>
      <a
        class="_all-link_5d4f08"
        href="/all"
      >
        View All
      </a>
    </div>
  </main>
</div>
`;
